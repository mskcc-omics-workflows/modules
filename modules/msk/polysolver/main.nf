process POLYSOLVER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://sachet/polysolver:v3':
        'docker.io/sachet/polysolver:v3' }"

    input:
    tuple val(meta), path(bam)
    val(includeFreq)
    val(build)
    val(format)
    val(insertCalc)

    output:
    tuple val(meta), path("*.hla.txt"), emit: hla
    path  "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args              = task.ext.args ?: ''
    def prefix            = task.ext.prefix ?: "${meta.id}"
    def includeFreq_param = includeFreq ?: 1
    def insertCalc_param  = insertCalc ?: 0
    def build_param       = build ? (["GRCh37","hg19"].contains(build) ? "hg19" : "hg38") : "hg19"
    def format_param      = format ?: "STDFQ"
    """
    cp /home/polysolver/scripts/shell_call_hla_type .
    sed -i "171s/TMP_DIR=.*/TMP_DIR=nf-scratch/" shell_call_hla_type
    bash shell_call_hla_type \\
        ${bam} \\
        Unknown \\
        ${includeFreq_param} \\
        ${build_param} \\
        ${format_param} \\
        ${insertCalc_param} \\
        ${prefix}

    mv ${prefix}/winners.hla.txt ${prefix}.hla.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        polysolver: v3
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.hla.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        polysolver: v3
    END_VERSIONS
    """
}
