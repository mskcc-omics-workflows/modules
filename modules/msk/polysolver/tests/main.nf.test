nextflow_process {

    name "Test Process POLYSOLVER"
    script "../main.nf"
    process "POLYSOLVER"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "polysolver"

    test("homo_sapiens - bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true)
                ]
                input[1] = []
                input[2] = []
                input[3] = []
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('test_paired_end_sorted_bam')
                ]
                input[1] = []
                input[2] = []
                input[3] = []
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
