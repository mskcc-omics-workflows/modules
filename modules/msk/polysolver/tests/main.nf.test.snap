{
    "homo_sapiens - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hla.txt:md5,34a1d26df0879e3dc4d36bd8067e4d69"
                    ]
                ],
                "1": [
                    "versions.yml:md5,60ce7b8e5d89d0b6ea3e01dc9a032fd6"
                ],
                "hla": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hla.txt:md5,34a1d26df0879e3dc4d36bd8067e4d69"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,60ce7b8e5d89d0b6ea3e01dc9a032fd6"
                ]
            }
        ],
        "timestamp": "2024-03-25T17:28:23.33869"
    },
    "homo_sapiens - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hla.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,60ce7b8e5d89d0b6ea3e01dc9a032fd6"
                ],
                "hla": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hla.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,60ce7b8e5d89d0b6ea3e01dc9a032fd6"
                ]
            }
        ],
        "timestamp": "2024-03-25T17:28:29.809368"
    }
}